### PR TITLE
Add support for uuid crate (v1)

### PR DESCRIPTION
**Pull request by merlleu**
Hello, 
I added support for the Uuid type from the uuid crate.
It is represented as a [u8; 16].

### CODE REVIEW

File: src/ext_uuid.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+impl< 'a, C > Readable< 'a, C > for Uuid
+    where C: Context
+{
+    #[inline]
+    fn read_from< R: Reader< 'a, C > >( reader: &mut R ) -> Result< Self, C::Error > {
+        let buffer: Vec<u8> = reader.read_vec(UUID_SIZE)?;
+        Ok(Uuid::from_bytes(
+            buffer.try_into()
+                .map_err(|_| crate::error::Error::custom( format!( "failed to read a uuid") ).into())?
+            ))
```
Comment (koute) [inline src/ext_uuid.rs:26]: This should use `read_bytes` instead of `read_vec`.

File: Cargo.toml [hunk 2]
```diff
@@ -18,6 +18,7 @@ chrono = { version = "0.4", optional = true }
 glam = { version = ">= 0.15, <= 0.20", optional = true }
 smallvec = { version = "1", optional = true }
 regex = { version = "1", optional = true, default-features = false }
+uuid = {version = "1", optional = true, default-features = false, features = ["v4"] }
```
Comment (koute) [inline Cargo.toml:21]: The `v4` feature should not be necessary as far as I can see?

File: tests/serialization_tests.rs [hunk 3]
```diff
@@ -2176,6 +2176,23 @@ fn test_regex() {
...
+
+    let uid = uuid::Uuid::parse_str("99ac1675-12ef-495b-bf56-606ebe5e3e71").unwrap();
+    let buffer = uid.write_to_vec().unwrap();
+    assert_eq!( buffer, &[0x99, 0xAC, 0x16, 0x75, 0x12, 0xEF, 0x49, 0x5B, 0xBF, 0x56, 0x60, 0x6E, 0xBE, 0x5E, 0x3E, 0x71] );
+
+    let deserialized = uuid::Uuid::read_from_buffer( &buffer ).unwrap();
+    assert_eq!( uid.to_string(), deserialized.to_string() );
+}
+
+
```
Comment (koute) [inline tests/serialization_tests.rs:2195]: Please delete this test and add a test in a similar manner to how the tests for `chrono`/`smallvec`/`indexmap` are.

**Review by koute — CHANGES_REQUESTED** [review]

**Comment by merlleu** [comment]
Hello, I made all your requested changes.

**Comment by koute** [comment]
Thanks!

**Pull request merged**